Foundation/NSCache.swift

```diff
@@ -20,15 +20,45 @@ private class NSCacheEntry<KeyType : AnyObject, ObjectType : AnyObject> {
     }
 }
 
+fileprivate class NSCacheKey: NSObject {
+    
+    var value: AnyObject
+    
+    init(_ value: AnyObject) {
+        self.value = value
+        super.init()
+    }
+    
+    override var hashValue: Int {
+        switch self.value {
+        case let nsObject as NSObject:
+            return nsObject.hashValue
+        case let hashable as Hashable:
+            return hashable.hashValue
+        default: return 0
+        }
+    }
+    
+    override func isEqual(_ object: Any?) -> Bool {
+        guard let other = (object as? NSCacheKey) else { return false }
+        
+        guard let left = self.value as? NSObject,
+            let right = other.value as? NSObject else { return self.value === other.value }
```

Comment:
Using `guard self.value !== other.value` for this check would probably be a bad style, isn't it?